projects/telemetry/src/nmea.c

```diff
@@ -0,0 +1,131 @@
+#include "nmea.h"
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include "log.h"
+#include "status.h"
+#include "util.h"
+
+// Just a function to losely validate if a sentence is valid
+StatusCode is_valid_nmea(const uint8_t *to_check, size_t len) {
+  if (len < 1 && to_check[0] != '$') {
+    LOG_DEBUG("First character of a NMEA sentence should be a $, it is a %c\n", to_check[0]);
+    return STATUS_CODE_UNKNOWN;
+  }
+  if (len < 3 && (to_check[1] != 'G' || to_check[2] != 'P')) {
+    LOG_DEBUG("NMEA sentence should begin with GP. They are %c%c\n", to_check[1], to_check[2]);
+    return STATUS_CODE_UNKNOWN;
+  }
+  return STATUS_CODE_OK;
+}
+
+StatusCode get_nmea_sentence_type(const uint8_t *rx_arr, NMEAMessageID *result) {
+  // Array index 3 should be 0, so that it is a null terminated string
+  uint8_t message_id[4] = { 0 };
```

Comment:
I think you improve this by replacing `message_id` with a uint32_t and a bit of bit manipulation. You could shift each of the characters into a 4-byte integer and then run a direct comparison with the message id values instead of using `strcmp()` to compare the entire string array.

Reply:
Will take a look at this, but it should be fine since `message_id` is at most a few characters.